miqi/agent/tools/filesystem.py

```
@@ -1156,11 +1327,37 @@ def parameters(self) -> dict[str, Any]:
+        for p in paths:
+            shared = await _resolve_write_shared_roots(
+                p,
+                base_dir=base_dir,
+                workspace_root=self._base_workspace or self._workspace,
+                shared=self._shared_roots,
+                granted=self._granted,
+                write_resolver=self._write_resolver,
+                persist_extra_root=self._persist_extra_root,
+            )
+            if shared is None:
+                return f"Error: 权限被拒绝：用户未授权写入 {p}"
```

Ruff (0.16.2)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)